File: src/simulation/RobotLeg.java
package simulation;

import org.jbox2d.dynamics.Body;
import org.jbox2d.dynamics.joints.RevoluteJoint;

import java.util.ArrayList;

/**
 * Created by colander on 15.3.17.
 * An abstraction of the legs of the robot
 */
public class RobotLeg {

    ArrayList<Body> segments;
    ArrayList<RevoluteJoint> joints;
    boolean touch = false;
    double touchValue = 0;

    public RobotLeg(ArrayList<Body> segments, ArrayList<RevoluteJoint> joints) {
        this.segments = segments;
        this.joints = joints;
    }
}

File: src/main/java/com/janboucek/crawler/simulation/RobotLeg.kt
package com.janboucek.crawler.simulation

import org.jbox2d.dynamics.Body
import org.jbox2d.dynamics.joints.RevoluteJoint

/**
 * Created by colander on 15.3.17.
 * An abstraction of a robot's legs
 */
data class RobotLeg(val segments: List<Body>, val joints: List<RevoluteJoint>) {
    var touch = false
    var touchValue = 0.0
}